File: stembranch.py
stem = "甲乙丙丁戊己庚辛壬癸"
branch = "子丑寅卯辰巳午未申酉戌亥"


def stembranch(year):
    return stem[(year - 4) % 10] + branch[(year - 4) % 12]


class StemBranch:
    stem = "甲乙丙丁戊己庚辛壬癸"
    branch = "子丑寅卯辰巳午未申酉戌亥"

    def __init__(self):
        pass

    def stembranch(self, year):
        return self.stem[(year - 4) % 10] + self.branch[(year - 4) % 12]

File: stembranch.r
stem <- c("甲", "乙", "丙", "丁", "戊", "己", "庚", "辛", "壬", "癸")
branch <- c("子", "丑", "寅", "卯", "辰", "巳", "午", "未", "申", "酉", "戌", "亥")
stembranch <- function(year) {
  return(paste0(stem[(year-3)%%10], branch[(year-3)%%12]))
}
